msfvenom

```diff
@@ -69,81 +69,69 @@ def parse_args(args)
   opt.separator('')
   opt.separator('Options:')
 
+  opt.on('-l', '--list            <type>', Array, 'List all module for [type]. Types are: payloads, encoders, nops, platforms, encrypt, formats, all') do |l|
+    if l.nil? or l.empty?
+      l = ["all"]
+    end
+    opts[:list] = l
+  end
+
   opt.on('-p', '--payload         <payload>', String,
-         'Payload to use. Specify a \'-\' or stdin to use custom payloads') do |p|
+         'Payload to use (--list payloads to list, --list-options for arguments). Specify \'-\' or STDIN for custom') do |p|
     if p == '-'
       opts[:payload] = 'stdin'
     else
       opts[:payload] = p
     end
   end
 
-  opt.on('--payload-options', "List the payload's standard options") do
+  opt.on('--list-options', "List --payload <value>'s standard, advanced and encryption options") do
```

Comment:
This should read

`List --payload <value>'s standard, advanced and evasion options`

Reply:
Agreed!